src/Facilitator/Facilitator.js

```diff
@@ -662,6 +662,136 @@ class Facilitator {
       });
   }
 
+  /**
+   * Perform redeem.
+   *
+   * @param {string} amount Redeem amount
+   * @param {string} beneficiary Beneficiary address.
+   * @param {string} gasPrice Gas price.
+   * @param {string} gasLimit Gas limit.
+   * @param {string} nonce Redeemer's nonce.
+   * @param {string} hashLock Hash lock;
+   * @param {Object} txOptions Transaction options.
+   * @return {Promise} Promise object
+   */
+  async redeem(amount, beneficiary, gasPrice, gasLimit, hashLock, txOptions) {
+    logger.info('Redeem');
+    logger.info('-----------------------');
+    if (new BN(amount).eqn(0)) {
+      const err = new TypeError('Redeem amount must not be zero.');
+      return Promise.reject(err);
+    }
+
+    if (!Web3.utils.isAddress(beneficiary)) {
+      const err = new TypeError('Invalid beneficiary address.');
+      return Promise.reject(err);
+    }
+
+    if (typeof gasPrice !== 'string') {
+      const err = new TypeError('Invalid gas price.');
+      return Promise.reject(err);
+    }
+
+    if (typeof gasLimit !== 'string') {
+      const err = new TypeError('Invalid gas limit.');
+      return Promise.reject(err);
+    }
+
+    if (typeof hashLock !== 'string') {
+      const err = new TypeError('Invalid hash lock.');
+      return Promise.reject(err);
+    }
+
+    if (!txOptions) {
+      const err = new TypeError('Invalid transaction options.');
+      return Promise.reject(err);
+    }
+
+    if (!Web3.utils.isAddress(txOptions.from)) {
+      const err = new TypeError('Invalid redeemer address.');
+      return Promise.reject(err);
+    }
+
+    const redeemer = txOptions.from;
+
+    logger.info('* Getting bounty amount *');
+    const bounty = await this.coGateway.getBounty().catch((exception) => {
+      logger.info('  - Exception while getting bounty amount');
+      return Promise.reject(exception);
+    });
+
+    if (!new BN(txOptions.value).eq(new BN(bounty))) {
+      logger.info(
+        '  - Value in transaction option is not equal to the bounty amount',
+      );
+      return Promise.reject(false);
+    }
+
+    logger.info(
+      '* Checking if redeemer has approved CoGateway for token transfer *',
```

Comment:
Where is it documented when to log the text, when to log starting `* `, and when to log starting `  - `?

Reply:
Hm ok. It's nice. We should probably agree on some kind of standard so that logging from different parts of the program doesn't look different when a user uses mosaic.js 😅

Reply:
Yes, agree. We can create a new ticket for this

Reply:
Added comment on #49